app/src/main/java/com/framgia/music_50/data/source/remote/GetTracksAsyncTask.java

```diff
@@ -0,0 +1,83 @@
+package com.framgia.music_50.data.source.remote;
+
+import android.os.AsyncTask;
+import com.framgia.music_50.data.model.Track;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import static com.framgia.music_50.utils.Constant.COLLECTION;
+import static com.framgia.music_50.utils.Constant.GET_METHOD;
+import static com.framgia.music_50.utils.Constant.REQUEST_TIMEOUT;
+import static com.framgia.music_50.utils.Constant.USER;
+
+public class GetTracksAsyncTask extends AsyncTask<String, Void, List<Track>> {
+    private OnFetchDataListener<Track> mListener;
+
+    public GetTracksAsyncTask(OnFetchDataListener<Track> listener) {
+        this.mListener = listener;
+    }
+
+    @Override
+    protected List<Track> doInBackground(String... strings) {
+        List<Track> tracks = null;
+        try {
+            String json = getDataFromUrl(strings[0]);
+            tracks = readDataFromJson(json);
+        } catch (IOException | JSONException e) {
+            mListener.onError(e);
+        }
+        return tracks;
+    }
+
+    @Override
+    protected void onPostExecute(List<Track> data) {
+        super.onPostExecute(data);
+        mListener.onSuccess(data);
+    }
+
+    private String getDataFromUrl(String s) throws IOException {
```

Comment:
khai báo tên có nghĩa
```suggestion
    private String getDataFromUrl(String url) throws IOException {
```